modules/finance/domain/aggregates/debt/debt_repository.go

```diff
@@ -23,12 +23,23 @@ type FindParams struct {
 	Status         *DebtStatus
 }
 
+type CounterpartyAggregate struct {
+	CounterpartyID                uuid.UUID
```

Comment:
Move to debt.go. Use an interface and follow existing codebase patterns with private implementations, functional options & constructors